commons-executors/src/main/java/com/palantir/common/concurrent/PTExecutors.java

```diff
@@ -127,6 +128,24 @@ public static ThreadPoolExecutor newCachedThreadPool(ThreadFactory threadFactory
     }
 
     /**
+     * Creates a thread pool that maintains between corePoolSize and maximumPoolSize threads. When maximumPoolSize
+     * is hit, new requests will not be queued but rather will be run in the caller's thread.
+     *
+     * @return the newly created thread pool
+     */
+    public static ThreadPoolExecutor newScalingThreadPoolWithCallerRunsOnEmpty(
```

Comment:
I'm a little hesitant to expose this publicly, might isolate it to C*KVS for now